tests/cacl/test_cacl_application.py

```diff
@@ -550,6 +552,10 @@ def generate_expected_rules(duthost, tbinfo, docker_network, asic_index, expecte
         # Default drop rules
         iptables_rules.append("-A INPUT -j DROP")
         ip6tables_rules.append("-A INPUT -j DROP")
+   
+    # IP Table rule to allow eth1-midplane traffic for chassis 
+    if asic_index is None:
```

Comment:
Do we need the if check here ? Looking at the implementation of append_midplane_rules, seems like it will be a no-op if the eth1-midplane device is not present.

Reply:
we need this check here because we need that rule in the expected rules list only when asic_index in None.
If the expected_rules list is created for a specific namespace, then we don't need the midplane rule added in the expected rule list.
On multi-asic Linecard:
```
admin@lc4-1:~$ sudo iptables -S | grep midplane
-A INPUT -i eth1-midplane -j ACCEPT
admin@lc4-1:~$ sudo ip netns exec asic0 iptables -S | grep midplane
admin@lc4-1:~$ 
```

Reply:
My only point was that given the implementation of the `append_midplane_rules` function, in the case where the midplane rule wasn't present, it seems like it would result in nothing being added to the expected list, hence it would be a no-op to call it without the if check.

It isn't a big deal obviously.

Reply:
No, this if condition will always be needed.
append_midplane_rules function will check if midplane rule is present in host namesapce and it will always be present in host namespace.

Reply:
Okay - I understand now. Thanks for clarifying.